### PR TITLE
make LocalRegistry work with the new breaking version of RegistryTools

### PR DESCRIPTION
**Pull request by KristofferC**
See https://github.com/JuliaRegistries/RegistryTools.jl/pull/78 for rationale.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,14 +4,17 @@ authors = ["Gunnar Farnebäck <gunnar@lysator.liu.se>"]
 version = "0.5.2"
 
 [deps]
+AutoHashEquals = "15f4f7f2-30c1-5605-9d31-71845cf9641f"
+CodecZlib = "944b1d66-785c-5afd-91f1-9de20f533193"
+Multibreak = "2db7fbaf-c681-4fb2-9fc5-9b884bdd742d"
```
Comment (GunnarFarneback) [inline Project.toml:9]: These are not package dependencies. At most test dependencies.

File: test/regression.jl [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 # Set up packages and registry directories in a temporary location.
 if VERSION >= v"1.2"
-    testdir = mktempdir(prefix = "LocalRegistryTests")
+    testdir = mktempdir(prefix = "LocalRegistryTests", cleanup=false)
```
Comment (GunnarFarneback) [inline test/regression.jl:18]: Debugging leftover?

File: test/utils.jl [hunk 3]
```diff
@@ -77,6 +77,8 @@ function compare_file_trees(path1, path2)
     if isdir(path1) && isdir(path2)
         dir1 = readdir_excluding_git(path1)
         if sort(dir1) != sort(readdir_excluding_git(path2))
+            @show sort(dir1)
+            @show sort(readdir_excluding_git(path2))
```
Comment (GunnarFarneback) [inline test/utils.jl:81]: Remove.

File: src/LocalRegistry.jl [hunk 4]
```diff
@@ -417,7 +417,7 @@ function find_registry_path(registry::AbstractString)
     return registry
 end
 
-function find_registry_path(::Nothing, pkg::Pkg.Types.Project)
+function find_registry_path(::Nothing, pkg::RegistryTools.Project)
```
Comment (GunnarFarneback) [inline src/LocalRegistry.jl:420]: Inconsistent to qualify the module only here.

**Comment by KristofferC** [comment]
Removed all the debug stuff, sorry abou that.

**Comment by KristofferC** [comment]
Ugh, do you think it is fine to drop julia 1.1?

**Comment by GunnarFarneback** [comment]
It's gonna be difficult to keep supporting Julia 1.1 when RegistryTools doesn't.

**Comment by KristofferC** [comment]
Well, we can make the code here compatible with both RegistryTools 1.9 and 2.0 (thereby supporting Julia 1.1 as well).

**Comment by GunnarFarneback** [comment]
That sounds exceedingly painful. Better give up on the pre-1.6 versions.

**Comment by GunnarFarneback** [comment]
Thanks.

**Pull request merged**